Add operations for setting state and basis state and integrate this with `skip_initial_state_prep` (#955)
PennyLaneAI/catalyst: doc/changelog.md

  Catalyst variant jaxpr.
  [(#837)](https://github.com/PennyLaneAI/catalyst/pull/837)

> * Allows for simulators to specify the optimization of setting the initial state directly
>   via `qml.StatePrep` and `qml.BasisState` when using the `initial_state_prep` flag in the
>   TOML file.

Reviewer: ```suggestion
* On devices that support it, initial state preparation routines `qml.StatePrep` and `qml.BasisState`
  are no longer decomposed when using Catalyst, improving compilation & runtime performance.
```

  `extrapolate_kwargs` keyword argument in `mitigate_with_zne`.
  [(#806)](https://github.com/PennyLaneAI/catalyst/pull/806)

> * The QuantumDevice API has now added the functions `SetState` and `SetBasisState`
> * The QuantumDevice API has now added the functions `SetState` and `SetBasisState`

Reviewer: Sorry I think this one bugged out
```suggestion
* The QuantumDevice API has now added the functions `SetState` and `SetBasisState`
```

  `extrapolate_kwargs` keyword argument in `mitigate_with_zne`.
  [(#806)](https://github.com/PennyLaneAI/catalyst/pull/806)

* The QuantumDevice API has now added the functions `SetState` and `SetBasisState`

>   for simulators that may benefit from instructions that directly set the state.

Reviewer: ```suggestion
* The QuantumDevice API has now added the functions `SetState` and `SetBasisState`
  for simulators that may benefit from instructions that directly set the state.
  Implementing these methods is optional, and device support can be indicated via
  the `initial_state_prep` flag in the TOML configuration file.
```
